src/compiler/checker.ts

```diff
@@ -11444,10 +11444,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             reportImplicitAny(element, anyType);
         }
         // When we're including the pattern in the type (an indication we're obtaining a contextual type), we
-        // use a non-inferrable any type. Inference will never directly infer this type, but it is possible
+        // use a non-inferrable unknown type. Inference will never directly infer this type, but it is possible
         // to infer a type that contains it, e.g. for a binding pattern like [foo] or { foo }. In such cases,
-        // widening of the binding pattern type substitutes a regular any for the non-inferrable any.
-        return includePatternInType ? nonInferrableAnyType : anyType;
+        // widening of the binding pattern type substitutes a regular any for the non-inferrable unknown.
+        return includePatternInType ? nonInferrableUnknownType : anyType;

```

Comment:
I'm not sure when "widening of the binding pattern type substitutes a regular any for the non-inferrable any" is happening and if this part of the comment is even relevant today.